lib/puppet/provider/package/gem.rb

```diff
@@ -122,7 +122,12 @@ def query
   end
 
   def uninstall
-    gemcmd "uninstall", "-x", "-a", resource[:name]
+    command = [command(:gemcmd), "uninstall"]
+    command << "-x" << "-a" << resource[:name]
```

Comment:
Could you expand these flags to make the command more clear? (`--executables` and `--all`)